src/js/tech/tech.js

```diff
@@ -531,20 +548,33 @@ class Tech extends Component {
    *
    * @param {Object} options The object should contain values for
    * kind, language, label and src (location of the WebVTT file)
+   * @param {Boolean} manualCleanup if set to false, the TextTrack will be
+   * automatically removed from the video element whenever the source changes
    * @return {HTMLTrackElement}
    * @method addRemoteTextTrack
    */
-  addRemoteTextTrack(options) {
+  addRemoteTextTrack(options, manualCleanup) {
     const track = mergeOptions(options, {
       tech: this
     });
-
     const htmlTrackElement = new HTMLTrackElement(track);
 
+    if (manualCleanup !== true && manualCleanup !== false) {
+      // deprecation warning
+      log.warn('Calling addRemoteTextTrack without explicitly setting the "manualCleanup" parameter to `true` is deprecated and default to `false` in future version of video.js');
```

Comment:
I am open to some nice word-smithing here. I feel like coming up with error/warnings is not my strong suit.